File: common/src/main/java/com/leo/commonutil/media/recode/IMediaRecordListener.java
package com.leo.commonutil.media.recode;

public interface IMediaRecordListener {
    /**
     * 录音开始
     */
    void onStart();

    /**
     * 录音中...
     *
     * @param db   当前声音分贝
     * @param time 录音时长
     */
    void onProgress(double db, long time);

    /**
     * 停止录音
     *
     * @param filePath 保存路径
     */
    void onStop(String filePath, long time);

    /**
     * 取消录音
     */
    void onCancel();
}

File: common/src/main/java/com/leo/commonutil/media/recode/IMediaRecordListener.kt
package com.leo.commonutil.media.recode

interface IMediaRecordListener {
    /**
     * 录音开始
     */
    fun onStart()

    /**
     * 录音中...
     *
     * @param db   当前声音分贝
     * @param time 录音时长
     */
    fun onProgress(db: Double, time: Long)

    /**
     * 停止录音
     *
     * @param filePath 保存路径
     */
    fun onStop(filePath: String?, time: Long)

    /**
     * 取消录音
     */
    fun onCancel()
}